pkg/reporter/reporter_test.go

```diff
@@ -199,46 +207,9 @@ var _ = PDescribe("Reporter", func() {
 
 		Expect(err).To(Succeed())
 		Expect(files).ToNot(BeEmpty())
-		Expect(len(files)).To(Equal(10))
-		for _, file := range files {
-			By(fmt.Sprintf("testing file %s", file))
-			Expect(file).To(BeAnExistingFile())
-
-			if !strings.Contains(file, "metadata") {
-				fileBytes, err := ioutil.ReadFile(file)
-				Expect(err).To(Succeed(), "file does not exist")
-				data := make(map[string]interface{})
-				err = json.Unmarshal(fileBytes, &data)
-				Expect(err).To(Succeed(), "file data did not parse to json")
-
-				id := func(element interface{}) string {
-					return "row"
-				}
-				Expect(data).To(MatchAllKeys(Keys{
-					"report_slice_id": Not(BeEmpty()),
-					"metrics": MatchElements(id, AllowDuplicates, Elements{
-						"row": MatchAllKeys(Keys{
-							"additionalLabels": MatchAllKeys(Keys{
-								"namespace": Equal("metering-example-operator"),
-								"pod":       Equal("example-app-pod"),
-								"service":   Equal("example-app-pod"),
-							}),
-							"domain":              Equal("apps.partner.metering.com"),
-							"interval_start":      HavePrefix("2020-"),
-							"interval_end":        HavePrefix("2020-"),
-							"kind":                Or(Equal("App"), Equal("App2")),
-							"report_period_end":   Equal(endStr),
-							"report_period_start": Equal(startStr),
-							"rhmUsageMetrics": MatchAllKeys(Keys{
-								"rpc_durations_seconds_count": BeAssignableToTypeOf(""),
-								"rpc_durations_seconds_sum":   BeAssignableToTypeOf(""),
-							}),
-							"version": Equal("v1"),
-						}),
-					}),
-				}))
-			}
-		}
+		Expect(len(files)).To(Equal(2), "length of files")
+
+		runTestOnFiles(files, expectedFields)
```

Comment:
This test is failing because of a race condition. If the additional fields tests run before, it sets the additionalFields memory.